homeassistant/components/fan/reproduce_state.py

```diff
@@ -56,37 +62,45 @@ async def _async_reproduce_state(
         )
         return
 
-    # Return if we are already at the right state.
-    if cur_state.state == state.state and all(
-        check_attr_equal(cur_state.attributes, state.attributes, attr)
-        for attr in ATTRIBUTES
-    ):
-        return
-
-    service_data = {ATTR_ENTITY_ID: state.entity_id}
-    service_calls = {}  # service: service_data
+    service_calls: list[tuple[str, dict[str, Any]]] = []
```

Comment:
I think the `service_calls` format can stay how it is as a dict, since dicts preserve insertion order by default.  This will just make the diff a bit smaller